guacamole-common-js/src/main/webapp/modules/Client.js

```diff
@@ -687,6 +687,20 @@ Guacamole.Client = function(tunnel) {
      *     A status object which describes the error.
      */
     this.onerror = null;
+    
+    /**
+     * Fired when a message is received from the remote tunnel and needs to be
+     * displayed to the user.
+     * 
+     * @event
+     * @param {!number} msgcode
+     *     A status code sent by the remote server that indicates the message
+     *     that should be displayed to the client.
+     * @param {string[]} args
+     *     An array of arguments that can be added to the message that will be
+     *     displayed to the client. 
+     */
```

Comment:
I think this should be documented more along the lines of receiving a message indicating some sort of arbitrary event or state change, not necessarily something that _needs_ or _will_ to be displayed to the user.